src/app/home/home.page.ts

```diff
@@ -68,23 +68,26 @@ export class HomePage {
     return new Promise((resolve, reject) => {
       this.countryService.getCountries().pipe(
         catchError(async (error) => {
-          this.handleError(error);
           reject(error);
+          this.handleError(error);
         }),
         tap(result => {
           if (result && !result.error) {
-            this.countries = result.data;
-            this.baseCountries = result.data;
-            this.groupCountries();
-            this.createLetters();
-            this.baseBackupGroupedCountries = { ...this.groupedCountries };
+            this.handleLoadCountriesSuccess(result.data);
             resolve();
           }
         })
       ).subscribe();
     });
 
   }
+  private handleLoadCountriesSuccess(data: Country[]): void {
```

Comment:
1- Direct Assignment: Assigns data directly to this.countries and this.baseCountries using the spread operator ([...data]).